core/src/Kokkos_View.hpp

```diff
@@ -264,14 +264,38 @@ class View : public Impl::BasicViewFromTraits<DataType, Properties...>::type {
     // base class doesn't have constraint
     // FIXME: Eventually we need to deprecate this behavior and just use
     // BasicView implementation
+#ifdef KOKKOS_ENABLE_DEPRECATED_CODE_4
+    using l_t = typename mdspan_type::layout_type;
+    if (r >= rank()) {
+      if constexpr (std::is_same_v<l_t, layout_right> ||
+                    Impl::IsLayoutRightPadded<l_t>::value) {
+        return 1;
+      }
+      if constexpr (std::is_same_v<l_t, layout_left> ||
+                    Impl::IsLayoutLeftPadded<l_t>::value) {
+        return stride(rank() - 1) * extent(rank() - 1);
+      }
+      if constexpr (std::is_same_v<l_t, layout_stride>) {
+        return 0;
+      }
+    }
+#endif
```

Comment:
```suggestion
#else
  KOKKO_ASSERT(r < rank());
#endif
```

Reply:
done

Reply:
Test that it fails?

Reply:
If that will be our new guarantee I think we should test for it to fail

Reply:
Add a DEATH_TEST or track so we don't forget